Refactor picker configuration flow

APetjko opened the pull request:
Setup picker via ChiliPhotoPicker
Save picker settings to PickerConfiguration object
Adjust layout to support landscape orientation
Convert PickerState to PickerViewModel to save state after config changed
Add possibility to limit maximal count of pickable images

APetjko committed:
Refactor picker configuration flow

Create picker via ChiliPhotoPickerBuilder
Save picker settings to PickerConfiguration object
Adjust layout to support landscape orientation
Convert PickerState to PickerViewModel to save state after config changed

APetjko committed:
Leave old picker creation way marked as Deprecated

APetjko committed:
Return old picker creation way, but move configuration to App

Define FileProviders' authority and ImageLoader implementation in App's onCreate
Pass Allow Multiple, Allow Camera and Theme via fragment arguments

APetjko committed:
Allow to limit max selection count

tairrzayev-chi commented on photopicker/src/main/java/lv/chi/photopicker/ChiliPhotoPicker.kt:
```diff
@@ -0,0 +1,18 @@
+    * @param authority - FileProvider's authority. Is necessary if you allow picker to use Camera,
+    *                    so it can store temporary files. Could be null otherwise
+    */
+    fun setUp(loader: ImageLoader, authority: String? = null) {
```
[Nitpick] Methods like this are usually called `init` / `initialize`

APetjko replied:
Done

tairrzayev-chi commented on photopicker/src/main/java/lv/chi/photopicker/PickerConfiguration.kt:
```diff
@@ -0,0 +1,20 @@
+    }
+
+    fun getImageLoader(): ImageLoader =
+        imageLoader ?: throw IllegalStateException("ImageLoader is null. You probably forget to call ChiliPhotoPicker.setUp()")
```
Could probably simplify the message down to `IllegalStateException("PickerConfiguration.setUp was not called")`

tairrzayev-chi commented on photopicker/src/main/java/lv/chi/photopicker/PickerConfiguration.kt:
```diff
@@ -0,0 +1,20 @@
+        imageLoader ?: throw IllegalStateException("ImageLoader is null. You probably forget to call ChiliPhotoPicker.setUp()")
+
+    fun getAuthority(): String =
+        authority ?: throw IllegalStateException("Authority is null. You probably forget to pass it to ChiliPhotoPicker.setUp()")
```
In`ChiliPhotoPicker.setUp`, I'd document the fact that there will be a crash if the camera is accessed without providing the  authority

APetjko replied:
Done

tairrzayev-chi commented on photopicker/src/main/java/lv/chi/photopicker/utils/SingleLiveEvent.kt:
```diff
@@ -0,0 +1,68 @@
+ *
+ * Note that only one observer is going to be notified of changes.
+ */
+class SingleLiveEvent<T> : MutableLiveData<T>() {
```
I'd also leave  a reference to the original source of this file

APetjko replied:
Done

APetjko committed:
Code cleanup

Add clarifying comment about exceptions
Add link to source of SingleLiveEvent
Rename setUp function of ChiliPhotoPicker to init

APetjko committed:
Update README
